libvast/src/expression.cpp

```diff
@@ -182,6 +182,10 @@ bool operator<(const expression& x, const expression& y) {
 
 // -- free functions -----------------------------------------------------------
 
+expression hoist(const expression& expr) {
+  return caf::visit(hoister{}, expr);
```

Comment:
Q: Why does this not take the expression by value and moves it into the hoister, if it creates a copy anyways?

This applies to the `normalize` and the other functions below as well.

Reply:
Valid point, but slightly tangential. Right now we have a consistent API. If we do the change, we should do it for all free functions.

Mind creating a story?

Reply:
Can do.